scala/scala-impl/src/org/jetbrains/plugins/hydra/HydraRepository.scala

```diff
@@ -0,0 +1,22 @@
+package org.jetbrains.plugins.hydra
+
+import org.jetbrains.plugins.hydra.settings.HydraApplicationSettings
+
+/**
+  * @author Maris Alexandru
+  */
+object HydraRepository {
+
+  def getRepositoryUrl: String = {
+    val url = HydraApplicationSettings.getInstance().hydraRepositoryUrl
+
+    if(url.endsWith("/")) url.dropRight(1) else url
+  }
+
+  def getRepositoryName: String = {
+    HydraApplicationSettings.getInstance().hydraRepositoryUrl.split("/") match {
```

Comment:
Better use the `java.net.URL`, URLs can be complicated and I'd rather be safe.

Reply:
Solved it and moved all Hydra repository related stuff to Hydra Application Settings.